File: src/main/java/io/github/jolkert/discordbotlink/jda/command/PingCommand.java
package io.github.jolkert.discordbotlink.jda.command;

import io.github.jolkert.discordbotlink.DiscordBotLink;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import net.dv8tion.jda.api.events.message.MessageReceivedEvent;
import org.jetbrains.annotations.NotNull;

public class PingCommand extends DiscordCommand
{
	public PingCommand()
	{
		this.name = "ping";
	}
	
	@Override
	public void execute(String[] args, MessageReceivedEvent context)
	{
		context.getChannel().sendMessage("Online!").queue();
		DiscordBotLink.Bot.primaryGuild = context.getGuild();
	}

	@Override
	public void executeSlashCommand(@NotNull SlashCommandInteractionEvent event)
	{

	}
}

File: src/main/kotlin/io/github/jolkert/discordbotlink/jda/command/PingCommand.kt
package io.github.jolkert.discordbotlink.jda.command

import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.events.message.MessageReceivedEvent
import net.dv8tion.jda.api.interactions.commands.build.Commands
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData

class PingCommand : Command()
{
	override val name = "ping"
	override val aliases = emptyArray<String>()
	override val slashCommandData: SlashCommandData = Commands.slash("ping", "Ping the bot")

	override fun executeTextCommand(event: MessageReceivedEvent, vararg args: String) =
		event.channel.sendMessage("Online!").queue()

	override fun executeSlashCommand(event: SlashCommandInteractionEvent) =
		event.reply("Online!").setEphemeral(true).queue()
}